### ISSUE TITLE
Bulk upload AIP results to seqr 

### ISSUE THREAD

**Issue by lynnpais** [broadinstitute/seqr#3319]
**Is your feature request related to a problem? Please describe.**
Need the ability to bulk upload AIP results into seqr and highlight variants with tag.

**Describe the solution you'd like**
Create the ability to upload a file with the AIP results. This can be added to the existing  External Analysis page in seqr. 
Create a new AIP tag with a hover over for metadata, which include the AIP categories and date. This tag should not be editable by users. If there is new data on a previously tagged AIP variant, the full history can be shown in the hover over.

As AIP uses codes for their categories, also add the following user friendly annotations - 
AIP category | seqr tag category
1 | 1 - ClinVar P/LP
2 | 2 - New gene
3 | 3 - LoF
4 | 4 - De novo
5 | 5 - SpliceAI
Support | Support

**Describe alternatives you've considered**
NA

**Additional context**
Discussed with Chrissy and in our sprint planning meeting.

<img width="1042" alt="Screen Shot 2023-04-11 at 11 33 55 AM" src="https://user-images.githubusercontent.com/37872186/231214287-03e229f5-98d5-413c-81a3-e5aee1e2486c.png">

**Comment by hanars** [comment]
Blocked on getting an example AIP file

**Comment by lynnpais** [comment]
We are considering having a higher and lower priority AIP tag. This will need additional discussions.

**Comment by MattWellie** [comment]
Hi all, I'd like to suggest this being broken into two issues. On the AIP side we'll generate a slimmed-down output file specifically for ingestion by Seqr. A proposed format for that would be -

```python
{
    seqr_family_id (str): {
        variant_id (str): { 
             Key (str): Value (str)
        }
    }
}
```

Typically those values will be

* Key - Name of Category/short description
* Value - Date category was first seen in the wild

AIP is designed to be extensible, and we don't want the category names/mapping between numerical and text descriptions to become a Seqr sync problem.

---

On the Seqr side we'd like the ability to upload this file and have the key-value contents displayed as tags on the corresponding family. I'm not familiar with the `External Analysis page` page, but that sounds like a good fit. 

AIP results are designed to be incremental, so I think we want submission of new data to `replace` previously added data for a given cohort, but I'm happy to have a discussion on the mechanics of that.

We don't currently have a `regular` and `high` priority AIP tag, but that's on the roadmap. In this scenario my thinking was that AIP would be responsible for an appropriately formatted pair of files, one for each `confidence`. If the External Analysis upload page had an option to submit one or the other, the parsing logic would be identical, with only the resulting tag/button text/colour different between the two options.

**Comment by MattWellie** [comment]
Following up on a Slack conversation, proposed format for a seqr integration file:

```
{
    "metadata": {
        "categories": {
            "1": "ClinVar Pathogenic",
            "2": "New Gene-Disease Association",
            "3": "High Impact Variant",
            "4": "de Novo",
            "5": "High SpliceAI Score",
            "pm5": "ACMG PM5 - missense in same residue as known pathogenic",
            "support": "High in Silico Scores"
        }
    },
    "results": {
        "CPG1": {
            "1-123456-CT-C": {
                "categories": [
                    "support"
                ],
                "support_vars": [
                    "1-4567890-G-A"
                ],
                "independent": false,
                "labels": []
            },
            "1-4567890-G-A": {
                "categories": [
                    "2"
                ],
                "support_vars": [
                    "1-123456-CT-C"
                ],
                "independent": false,
                "labels": []
            },
        },
        "CPG2": {
            "17-1054710-CT-C": {
                "categories": [
                    "1", 
                    "3"
                ],
                "support_vars": [],
                "independent": true,
                "labels": [
                     "high_priority"
                 ]
            }
        }
    }
}
```

A Helper script exists in the AIP repo to parse the results of a run into this minimised format

https://github.com/populationgenomics/automated-interpretation-pipeline/blob/main/helpers/minimise_output_for_seqr.py

**Comment by hanars** [comment]
still blocked, an agreed upon file format is not the same as an actual file to test loading

**Comment by MattWellie** [comment]
Could I have a little input here please? If you need an actual file could you let me know which dataset to produce one for - presumably this has to be samples/variants present in your instance

**Comment by hanars** [comment]
There is no motion on this ticket. All updates to the ticket were just an internal miscommunication. Please continue as before, and if you have questions about what AIP data our team is expecting, continue to reach out to whoever was doing it before. This is an internal ticket to track feature development. I will be closing it out to prevent future miscommunication
